warmrestart/warmRestartHelper.h

```diff
@@ -0,0 +1,132 @@
+#ifndef __WARMRESTART_HELPER__
+#define __WARMRESTART_HELPER__
+
+
+#include <vector>
+#include <map>
+#include <unordered_map>
+
+#include "dbconnector.h"
+#include "producerstatetable.h"
+#include "netmsg.h"
+#include "table.h"
+#include "tokenize.h"
+#include "warm_restart.h"
+
+
+namespace swss {
+
+
+/* FieldValueTuple functor to serve as comparator for restorationMap */
+struct fvComparator
+{
+    bool operator()(const std::vector<FieldValueTuple> &left,
+                    const std::vector<FieldValueTuple> &right) const
+    {
+        /*
+         * The sizes of both tuple-vectors should always match within any given
+         * application, otherwise we are running into some form of bug.
+         */
+        assert(left.size() == right.size());
```

Comment:
When comparing two vectors,  how to ensure they are of same size? 
Say, a route has two fields in usual case: nexthop and intf,   what if for some routes, the third fields "blackhole" is set?

```
### ROUTE_TABLE
    ;Stores a list of routes
    ;Status: Mandatory
    key           = ROUTE_TABLE:prefix
    nexthop       = *prefix, ;IP addresses separated “,” (empty indicates no gateway)
    intf          = ifindex? PORT_TABLE.key  ; zero or more separated by “,” (zero indicates no interface)
    blackhole     = BIT ; Set to 1 if this route is a blackhole (or null0)
```

Reply:
Good point. For routing-app there's this "blackhole" corner-case which brakes the uniformity for routing schema. Basically we are allowing a route to be defined as: "prefix/len | blackhole = true". No other app/scenarios that i'm aware of allow this inconsistency. In my humble opinion, we should keep uniformity by allowing the common schema to be re-utilized as this: "prefix/len | nh: blackhole | if: null". This approach will also simplify the addition of new attributes in the future (AFI, SAFI, etc).

Also, if you look at orchagent you'll see that we are not passing down "blackhole" routes -- they are being eliminated right there in routeOrch. The only blackhole route being pushed down to ASIC_DB is the default route, and this one is not even exposed to AppDB. So in reality, no one is really consuming these black-hole schema.

In my opinion, I think this assert() should be left here as it prevents real problems down the road. And concerning the blackhole routes, i would like to make a very minor adjustment in its representation to avoid this uniformity i talked about.